typeclasses/npcshop/npcshop.py

```diff
@@ -257,13 +258,16 @@ def at_object_creation(self):
         self.cmdset.add_default(StoreRoomCmdSet)
 
     def return_appearance(self, caller):
-        wares = []
-        for item in self.contents:
-            if item.db.value:
-                wares.append(item.key)
+        wares = [item for item in self.contents if item.db.value]
         desc = super(StoreRoom, self).return_appearance(caller)
         for line in desc.split("\n"):
-            if line.strip().split("(")[0] in wares:
-                line = "|y" + line
+            if any(line.find(ware.key) > -1 for ware in wares):
+                try:
+                    line = "|y{line} |n({price})".format(
```

Comment:
I think popping the items off the wares list might be a little risky. How about something like
```:python
for line in desc.split("\n"):
    for ware in wares:
        if line.find(ware.get_display_name(caller)) > -1:
            line = "|y{item} |w{price}".format(item=ware.get_display_name(caller), 
                                               price=format_coin(ware.db.value))
```
Then again, it seems like there's always going to be the chance of getting mismatches, just from the nature of searching through the parent return_appearance output for matches. Non-builder characters can `recog` anything as anything else, and giving multiple items the same recog names would confuse my sample code and possibly just not be highlighted in your current branch code.

I wonder if it might just be easier to re-implement `return_appearance` by grabbing the parent implementation, and doing your own sorting of exits, **wares**, characters, and other items, and constructing the appearance string, rather than modifying what comes out of the parent implementation.